validator/client/src/main/java/tech/pegasys/artemis/validator/client/DepositSimulation.java

```diff
@@ -76,19 +70,56 @@ public String toJSON() {
                   "merkle_tree_index", deposits.get(i).getMerkle_tree_index().toHexString());
               arr.add(deposit);
             });
+    return arr;
+  }
+
+  @Override
+  public void filterOutputFields(List<String> outputFields) {
 
-    obj.add("events", arr);
+    for (String field : outputFields) {
+      switch (field) {
+        case "secp":
+          this.outputFieldMap.put(
+              "secp", validator.getSecpKeys().secretKey().bytes().toHexString());
+          break;
+
+        case "bls":
+          this.outputFieldMap.put(
+              "bls", validator.getBlsKeys().secretKey().toBytes().toHexString());
+          break;
+
+        case "deposit_data":
+          this.outputFieldMap.put("deposit_data", deposit_data.toHexString());
+          break;
+
+        case "events":
+          this.outputFieldMap.put("events", depositEvents());
+          break;
+      }
+    }
+  }
+
+  @Override
+  public String toJSON() {
+    Gson gson = new GsonBuilder().create();
+    GsonBuilder gsonBuilder = new GsonBuilder();
+    String jsonString = gson.toJson(this.outputFieldMap);
+    JsonObject deposit = gson.fromJson(jsonString, JsonObject.class);
     Gson customGson = gsonBuilder.setPrettyPrinting().create();
-    return customGson.toJson(obj);
+    return customGson.toJson(deposit);
   }
 
   @Override
   public String toCSV() {
-    return null;
+    String csvOutputString = "";
+    for (Object obj : this.outputFieldMap.values()) {
+      csvOutputString += "'" + obj.toString() + "',";
+    }
+    return csvOutputString.substring(0, csvOutputString.length() - 1);
   }
```

Comment:
Need a check to ensure that csvOutputString.length() > 0

Reply:
Done. please review.

Reply:
👍